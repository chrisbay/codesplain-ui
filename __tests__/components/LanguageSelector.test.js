import React from 'react'
import ReactTestUtils from 'react-addons-test-utils';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import LanguageSelector from '../../src/components/LanguageSelector'

const mockLanguages = [
  {text: "French", value: 'fr'},
  {text: "Japanese", value: 'jp'}
]

describe('<LanguageSelector />', () => {
  it('matches snapshot', () => {
    const renderer = ReactTestUtils.createRenderer();
    const tree = renderer.render(
      <MuiThemeProvider>
        <LanguageSelector
          languages={mockLanguages}
          onChange={jest.fn()}
          selected={mockLanguages[0].value}
        />
      </MuiThemeProvider>
    );
    expect(tree).toMatchSnapshot();
  })
})
