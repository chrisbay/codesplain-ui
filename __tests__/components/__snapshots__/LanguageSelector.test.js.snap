exports[`<LanguageSelector /> matches snapshot 1`] = `
<LanguageSelector
  languages={
    Array [
      Object {
        "text": "French",
        "value": "fr",
      },
      Object {
        "text": "Japanese",
        "value": "jp",
      },
    ]
  }
  onChange={[Function]}
  selected="fr" />
`;
